docs/JestObjectAPI.md

```diff
@@ -663,6 +663,22 @@ test('isLocalhost returns false when HOSTNAME is not localhost', () => {
 });
 ```
 
+Jest will make sure that the target property exists on the object and its value is not `undefined`. To skip this precaution, use `tolerateUndefined` option:
```

Comment:
I don't really like the name `tolerateUndefined`. I don't have any better suggestions off the top of my head, tho.

`tolerateMissing`, `allowMissing`, `setMissing`... neither sounds good, but `tolerates` is even worse imo 😅

Reply:
What about `respectUndefined`? Not sure. Hm.. Did you see my comment on non-writable values? What about option `strict: false` which would allow creating undefined props and overwriting non-writable values? Simple thing.

Reply:
Does `replaceProperty` actually disallow existing but `undefined` properties? I would consider this a bug.

And for non-existing properties, it probably makes more sense to add `jest.addProperty` instead which would act like `replaceProperty` + `tolerateUndefined`.

Reply:
Thanks for thinking together. How do you see replacement of non-writeable props? Should it just work or do we need an option?

Reply:
I don't think I want `addProperty` - then I think we'd need a `removeProperty` as well. It might have been better for `replaceProperty` to have been called `setProperty`, then taken options for replacement strategy. Anyways, I think a single API on `jest` is enough.